### PR TITLE
Feature/#11 avro nested data support

### LINKED ISSUE #5: Improve code coverage
Some of the methods are not covered with unit tests.

- Add more unit tests
- Refactor the connection object parsing

### LINKED ISSUE #11: Support Avro complex types
## Acceptance Criteria

- Avro Array, Map and Record types are stored as JSON strings

### PR DESCRIPTION
**Pull request by morazow**
Fixes #11.
Fixes #5.

### CODE REVIEW

**Review by AnastasiiaSergienko — CHANGES_REQUESTED** [review]
Just a few small comments

File: src/main/scala/com/exasol/avro/AvroRow.scala [hunk 1]
```diff
@@ -61,16 +85,16 @@ object AvroRow {
...
       case 2 =>
         if (types.get(0).getType() == Schema.Type.NULL) {
-          getAvroRecordValue(value, types.get(1))
+          getAvroValue(value, types.get(1))
         } else if (types.get(1).getType() == Schema.Type.NULL) {
-          getAvroRecordValue(value, types.get(0))
+          getAvroValue(value, types.get(0))
         } else {
           throw new IllegalArgumentException(
             "Avro Union type should contain a primitive and null!"
```
[the remark this answers is not in the document]
Comment (morazow) [reply src/main/scala/com/exasol/avro/AvroRow.scala:100]: Yes, union are represented as array of two, where one of them is null.
But it is about the union type so, `a primitive and null types` would be better.

File: src/main/scala/com/exasol/avro/AvroRow.scala [hunk 2]
```diff
@@ -15,40 +19,60 @@ import org.apache.avro.util.Utf8
...
     val fields = avroRecord.getSchema().getFields()
     for { index <- 0 until fields.size } {
-      values.update(index, getAvroRecordValue(avroRecord.get(index), fields.get(index).schema))
+      val fieldSchema = fields.get(index).schema()
+      val fieldValue = getAvroValue(avroRecord.get(index), fieldSchema)
+      if (isPrimitiveAvroType(fieldSchema.getType())) {
+        values.update(index, fieldValue)
+      } else {
+        values.update(index, JsonMapper.toJson(fieldValue))
+      }
```
[the remark this answers is not in the document]
Comment (morazow) [reply src/main/scala/com/exasol/avro/AvroRow.scala]: Done

**Review by AnastasiiaSergienko — APPROVED** [review]
:+1:

**Pull request merged**